src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1124,6 +1125,15 @@ void BlockchainLMDB::open(const std::string& filename, const int mdb_flags)
 
   size_t mapsize = DEFAULT_MAPSIZE;
 
+  if (db_flags & DBF_FAST)
+    mdb_flags |= MDB_NOSYNC;
+  if (db_flags & DBF_FASTEST)
+    mdb_flags |= MDB_NOSYNC | MDB_WRITEMAP | MDB_MAPASYNC;
+  if (db_flags & DBF_RDONLY)
+    mdb_flags = MDB_RDONLY;
```

Comment:
Right. None of the sync flags mean anything for a read-only session. Also, this turns off the noreadahead flag. Since blockchain_export is the one using RDONLY, we actually want readahead in this case.